recipes/google-cloud-cpp/all/conanfile.py

```diff
@@ -0,0 +1,276 @@
+            raise ConanInvalidConfiguration("Building requires VS >= 2019")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+
+    def requirements(self):
+        self.requires('protobuf/3.15.5')
+        self.requires('grpc/1.37.1')
+        self.requires('nlohmann_json/3.9.1')
+        self.requires('crc32c/1.1.1')
+        self.requires('abseil/20210324.0')
+        self.requires('libcurl/7.75.0')
+        self.requires('openssl/1.1.1k')
+        # TODO: Add googleapis once it is available in CCI (now it is embedded)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = 0
+
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK"] = False
+
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_BIGTABLE"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_BIGQUERY"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_SPANNER"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_STORAGE"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_FIRESTORE"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_PUBSUB"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_IAM"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_LOGGING"] = True
+        self._cmake.definitions["GOOGLE_CLOUD_CPP_ENABLE_GENERATOR"] = True
+
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, 'lib', "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'pkgconfig'))
+
+    def package_info(self):
+        self.cpp_info.components["experimental-bigquery"].requires = ["grpc_utils", "common", "cloud_bigquery_protos"]
+        self.cpp_info.components["experimental-bigquery"].libs = ["google_cloud_cpp_bigquery"]
+
+        self.cpp_info.components["bigtable"].requires = ["abseil::absl_memory", "bigtable_protos", "common", "grpc_utils", "grpc::grpc++", "grpc::grpc", "protobuf::libprotobuf"]
+        self.cpp_info.components["bigtable"].libs = ["google_cloud_cpp_bigtable"]
+
+        self.cpp_info.components["common"].requires = ["abseil::absl_any", "abseil::absl_flat_hash_map", "abseil::absl_memory", "abseil::absl_optional", "abseil::absl_time"]
+        self.cpp_info.components["common"].libs = ["google_cloud_cpp_common"]
```

Comment:
Generic component names can lead to conflicts with some generators

```suggestion
        self.cpp_info.components["google-cloud-common"].requires = ["abseil::absl_any", "abseil::absl_flat_hash_map", "abseil::absl_memory", "abseil::absl_optional", "abseil::absl_time"]
        self.cpp_info.components["google-cloud-common"].libs = ["google_cloud_cpp_common"]
        self.cpp_info.components["google-cloud-common"].names["cmake_find_package"] = "common"
        self.cpp_info.components["google-cloud-common"].names["cmake_find_package_multi"] = "common"
```

Reply:
👍